From 19600eb75aa9b1df3e4b0a4e55a5d08b957e1fd9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Mon, 13 May 2019 10:13:24 +0200
Subject: [PATCH 3/4] drivers/video: drop framebuffer size constraints
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

The limit 1900x1200 do not match real world devices (1900 looks like a
typo, should be 1920). But in practice the limits are arbitrary and do
not serve any real purpose. As discussed in "Increase framebuffer size
to todays standards" thread, drop them completely.

This fixes graphic console on device with 3840x2160 native resolution.

Suggested-by: Jan Beulich <jbeulich@suse.com>
Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>
---
 xen/drivers/video/lfb.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/xen/drivers/video/lfb.c b/xen/drivers/video/lfb.c
index 0475a68296..5022195ae5 100644
--- a/xen/drivers/video/lfb.c
+++ b/xen/drivers/video/lfb.c
@@ -10,9 +10,6 @@
 #include "lfb.h"
 #include "font.h"
 
-#define MAX_XRES 1900
-#define MAX_YRES 1200
-
 struct lfb_status {
     struct lfb_prop lfbp;
 
@@ -146,13 +143,6 @@ void lfb_carriage_return(void)
 
 int __init lfb_init(struct lfb_prop *lfbp)
 {
-    if ( lfbp->width > MAX_XRES || lfbp->height > MAX_YRES )
-    {
-        printk(XENLOG_WARNING "Couldn't initialize a %ux%u framebuffer early.\n",
-               lfbp->width, lfbp->height);
-        return -EINVAL;
-    }
-
     lfb.lfbp = *lfbp;
 
     lfb.lbuf = xmalloc_bytes(lfb.lfbp.bytes_per_line);
-- 
2.20.1
